### PR TITLE
Include portable .pdb files in regular nupkgs when Internal.AspNetCore.Sdk is used

### PR DESCRIPTION
**Pull request by natemcmaster**
Allows adding .pdb files in the .nupkg for aspnetcore packages that end up on NuGet.org.

cref https://github.com/aspnet/Universe/issues/131

cc @Eilon @JunTaoLuo @ctaggart

### CODE REVIEW

File: src/Internal.AspNetCore.Sdk/build/Common.targets [hunk 1]
```diff
@@ -15,6 +15,16 @@ For single-tfm projects, this will be imported from build/Internal.AspNetCore.Sd
     <IncludeSymbols Condition="'$(NuspecFile)'!=''">false</IncludeSymbols>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(IncludeSymbols)' != 'false' ">
+    <!-- Include .pdb files in regular .nupkg files. -->
+    <AllowedOutputExtensionsInPackageBuildOutputFolder>$(AllowedOutputExtensionsInPackageBuildOutputFolder);.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
+  </PropertyGroup>
+
+  <Target Name="_EnsurePdbsInNupkgArePortable" BeforeTargets="GenerateNuspec" Condition=" '$(IsPackable)' != 'false' ">
+    <Error Text="Symbols included in a nupkg package should be portable. Set DebugType=portable or IncludeSymbols=false."
```
Comment (pranavkm) [inline src/Internal.AspNetCore.Sdk/build/Common.targets]: Set DebugType=portable ~~or~~and IncludeSymbols=false

File: src/Internal.AspNetCore.Sdk/build/Common.targets [hunk 2]
```diff
@@ -15,6 +15,16 @@ For single-tfm projects, this will be imported from build/Internal.AspNetCore.Sd
...
+      Condition=" '$(DebugType)' != 'portable' AND '$(IncludeSymbols)' == 'true' " />
```
Comment (pranavkm) [inline src/Internal.AspNetCore.Sdk/build/Common.targets]: Btw, the last I looked, symbols for Razor's net46 binaries are `full`. Can you verify this works for the repo?

File: src/Internal.AspNetCore.Sdk/build/Common.targets (hunk 1 above)
Comment (natemcmaster) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: Maybe I should clarify. What I meant here was `Fix by setting DebugType=portable or disable inclusion of symbols by setting IncludeSymbols=false`

File: src/Internal.AspNetCore.Sdk/build/Common.targets (hunk 2 above)
Comment (natemcmaster) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: Why are they `full`? Don't we need `portable` for these to work for x-plat?

Comment (pranavkm) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: I'm not entirely sure. @NTaylorMullen might know more.

Comment (natemcmaster) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: Looks like it's due to this, right? https://github.com/aspnet/Razor/issues/1548 If we need `full` for VS, let's still produce `portable` by default. We can convert `portable` symbols to `full` in our VS publishing steps. I believe @muratg knows how to do that.

File: src/Internal.AspNetCore.Sdk/build/Common.targets (hunk 1 above)
Comment (pranavkm) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: Maybe split this in to two? 

> Symbols were set to be included but the project was built using `DebugType = full`. Set `DebugType = portable` to include portable pdbs required for symbol publishing or stop including symbols by setting `IncludeSymbols=false`

Comment (natemcmaster) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: that's much better. I'll update the error.

File: src/Internal.AspNetCore.Sdk/build/Common.targets (hunk 2 above)
Comment (pranavkm) [reply src/Internal.AspNetCore.Sdk/build/Common.targets]: Ahh, yeah. Completely forgot I made that change 👍

**Pull request merged**